lib/gh_proxy/github.ex

```diff
@@ -0,0 +1,40 @@
+defmodule GHProxy.GithubAPI do
+  alias GHProxy.User, as: User
+
+  @callback get_user(binary()) :: {:ok, User.t()} | {:error, String.t()}
+
+  def get_user(username), do: impl().get_user(username)
+  def impl, do: Application.get_env(:gh_proxy, :github, GHProxy.GithubImpl)
```

Comment:
Если вы не собираетесь менять имплементацию кэша в рантайме (что вряд ли), лучше так:
```suggestion
  @impl Application.compile_env(:gh_proxy, :github, GHProxy.GithubImpl)
  def impl, do: @impl
```

И да, я знаю, что это копипаста из документации `Mox`, но в документации оно так, чтобы можно было разные моки для разных тестов прикручивать (я хрен пойми зачем).